### PR TITLE
Add OpenNet install.sh

### PR DESCRIPTION
**Pull request by pichuang**
Support Ubuntu 14.04.1, CentOS 7, Fedora 21.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+function mininet {
+
+    echo "Fetch Mininet"
+    cd $ROOT_PATH
+    if [ ! -d mininet ]; then
+        git clone https://github.com/mininet/mininet.git
+    fi
+
+    cd $ROOT_PATH/mininet && git checkout tags/$MININET_VERSION
+    cp $ROOT_PATH/mininet-patch/util/install.sh $ROOT_PATH/mininet/util/
```
Comment (rascov) [inline install.sh]: Suggest use a patch instead of replace the entire install.sh

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+#title           : install.sh
+#description     : This script will install OpenNet
+#                  Support Ubuntu 14.04.1, CentOS 7, Fedora 21
+#==============================================================================
+
+set -o nounset
+set -e
+
+ROOT_PATH=`pwd`
+OVS_RELEASE='2.3.1'
```
Comment (rascov) [inline install.sh]: Suggest we change the variable name to OVS_VERSION.

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        /etc/init.d/openvswitch start
+    fi
+
+}
+
+function ns3 {
+
+    echo "Fetch ns-3.21"
+    cd $ROOT_PATH
+    if [ ! -f ns-allinone-3.21.tar.bz2 ]; then
```
Comment (rascov) [inline install.sh]: Suggest we use NS3_VERSION to replace the fixed version number.

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        curl -O -k https://www.nsnam.org/release/ns-allinone-3.21.tar.bz2
```
Comment (rascov) [inline install.sh]: Same comment as above.

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+    fi
+    tar xf ns-allinone-3.21.tar.bz2
+
+}
+
+function pygccxml {
+
+    echo "Fetch and install pygccxml"
+    cd $ROOT_PATH
+    if [ ! -f pygccxml-1.0.0.zip ]; then
```
Comment (rascov) [inline install.sh]: Suggest we use PYGCCXML_VERSION to replace the fixed version number.

File: install.sh [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+}
+
+
+function usage {
+
+    echo
+    echo Usage: $(basename $0) -[$PARA] >&2
+    echo "-a: Install OpenNet World"
+    echo "-f: Finish message"
+    echo
```
Comment (rascov) [inline install.sh]: Complete the usage of other options

**Comment by rascov** [comment]
Also, please run the installation script as a part of continuous integration.

File: install.sh (hunk 1 above)
Comment (pichuang) [inline install.sh]: 1. This work just that CentOS can work on the mininet.
2. Patch conditional statement about distribution support.
3. I will follow with you suggest, thx.

**Comment by rascov** [comment]
Manually merge into master

**Pull request closed without merging**